modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -84,9 +89,17 @@ public String getHelp() {
     public void processOpts() {
         super.processOpts();
 
-        if (additionalProperties.containsKey(USE_RX_JAVA)) {
+        if (additionalProperties.containsKey(USE_RX_JAVA) && additionalProperties.containsKey(USE_RX_JAVA2)) {
+            LOGGER.warn("You specified both RxJava versions 1 and 2 but they are mutually exclusive. Defaulting to v2.");
+        } else if (additionalProperties.containsKey(USE_RX_JAVA)) {
             this.setUseRxJava(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA).toString()));
         }
+        if (additionalProperties.containsKey(USE_RX_JAVA2)) {
+            this.setUseRxJava2(Boolean.valueOf(additionalProperties.get(USE_RX_JAVA2).toString()));
+        }
+	if (!useRxJava && !useRxJava2) {
+	    additionalProperties.put(DO_NOT_USE_RX, true);
+	}
```

Comment:
Also shall we replace these 3 lines with the following 1-liner?
```
	    additionalProperties.put(DO_NOT_USE_RX, doNotUseRx);
```
to ensure additional property has the latest `doNotUseRx` value (true or false)